### PR TITLE
Make the gerrit API client limit concurrency per instance, default to 5.

### PR DESCRIPTION
**Pull request by cjwagner**
Also start logging the headers of error responses from the API since these will be helpful for debugging.

/assign @listx

### CODE REVIEW

**Comment by cjwagner** [comment]
I'm not sure if 5 is a reasonable default or not. Open to suggestions.

File: prow/gerrit/client/client.go [hunk 1]
```diff
@@ -399,13 +406,15 @@ func (c *Client) QueryChangesForProject(instance, project string, lastUpdate tim
 
 func (c *Client) GetChange(instance, id string, addtionalFields ...string) (*ChangeInfo, error) {
 	c.lock.RLock()
-	defer c.lock.RUnlock()
 	h, ok := c.handlers[instance]
+	c.lock.RUnlock()
```
Comment (listx) [inline prow/gerrit/client/client.go:410]: Nice simplifying change.

**Comment by listx** [comment]
> I'm not sure if 5 is a reasonable default or not. Open to suggestions.

I was thinking the same. I think the right way might be to define a "max" concurrency value instead (something like 10? or 20?) and then make the client build up to it dynamically. We could track the number of 429s we get and throttle it down accordingly. But I think a static value of 5 is a good starting point.

File: prow/gerrit/client/client.go [hunk 2]
```diff
@@ -518,12 +535,15 @@ func (c *Client) Account(instance string) (*gerrit.AccountInfo, error) {
 
 func (c *Client) GetMergeableInfo(instance, changeID, revisionID string) (*gerrit.MergeableInfo, error) {
 	c.lock.RLock()
-	defer c.lock.Unlock()
```
Comment (cjwagner) [inline prow/gerrit/client/client.go:521]: By the way This was a bug. `RUnlock()` is the correct call.

**Pull request merged**